### PR TITLE
#777 os todo delete outbound lines

### PR DESCRIPTION
**Pull request by joshxg**
#777 

- Adds delete functionality for lines.
- Not exactly what to do about errors - at the moment, ignoring them 🤔 
- Also might want a confirmation dialog, but i'm unsure

### CODE REVIEW

File: packages/invoices/src/OutboundShipment/api/hooks.ts [hunk 1]
```diff
@@ -141,3 +146,92 @@ export const useSaveOutboundLines = () => {
...
+      return { previous, ids };
+    },
+    onError: (_, __, context) => {
+      queryClient.setQueryData(['invoice', id], context?.previous);
+    },
+    onSettled: () => {
+      queryClient.invalidateQueries(['invoice', id]);
+    },
+  });
+};
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/api/hooks.ts:186]: this is the actual mutation hook, which is the same as inbound, optimistically deleting

File: packages/invoices/src/OutboundShipment/api/hooks.ts [hunk 2]
```diff
@@ -141,3 +146,92 @@ export const useSaveOutboundLines = () => {
...
+        onSuccess,
+      });
+    } else {
+      const infoSnack = info(t('label.select-rows-to-delete-them'));
+      infoSnack();
+    }
+  };
+
+  return { onDelete };
+};
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/api/hooks.ts:237]: Unsure if this is wise to just throw it all into a single hook, but is convenient.. probably doing a few too many things..

File: packages/invoices/src/OutboundShipment/api/api.ts [hunk 3]
```diff
@@ -112,6 +111,12 @@ const invoiceToInput = (
   theirReference: patch.theirReference,
 });
 
+const getCreateDeleteOutboundLineInput =
```
Comment (mark-prins) [inline packages/invoices/src/OutboundShipment/api/api.ts:114]: such a good name!

File: packages/invoices/src/OutboundShipment/api/hooks.ts [hunk 4]
```diff
@@ -141,3 +146,92 @@ export const useSaveOutboundLines = () => {
...
+
+  const onDelete = async () => {
+    if (selectedRows && selectedRows?.length > 0) {
+      const number = selectedRows?.length;
+      const onSuccess = success(t('message.deleted-lines', { number }));
+      mutate(selectedRows, {
+        onSuccess,
+      });
+    } else {
+      const infoSnack = info(t('label.select-rows-to-delete-them'));
```
Comment (mark-prins) [inline packages/invoices/src/OutboundShipment/api/hooks.ts:231]: oh dear. the font size on the snack has gone crazy 🙄

File: packages/invoices/src/OutboundShipment/api/hooks.ts (hunk 2 above)
Comment (mark-prins) [reply packages/invoices/src/OutboundShipment/api/hooks.ts:237]: maybe, but it is convenient. and the selection of lines is good to have here

**Review by mark-prins — APPROVED** [review]
👍  Looks good!

File: packages/invoices/src/OutboundShipment/api/hooks.ts (hunk 4 above)
Comment (joshxg) [reply packages/invoices/src/OutboundShipment/api/hooks.ts:231]: Yeah. Might make an issue to start collecting a list of minor style issues?

File: packages/invoices/src/OutboundShipment/api/api.ts (hunk 3 above)
Comment (joshxg) [reply packages/invoices/src/OutboundShipment/api/api.ts:114]: oh.. ha!

**Pull request merged**